File: 38-Python/Caravan.py
from Veicolo import Veicolo

class Caravan(Veicolo):
    def __init__(self, codice, marca, larghezza, lunghezza, costo, posti, potenza) -> None:
        super().__init__(codice, marca, larghezza, lunghezza, costo, posti)
        self._potenza = potenza

    def __str__(self) -> str:
        return "caravan\t" + super().__str__() + "\t-\t-\t" + str(self._potenza) + "\t" + str(self._costo)

File: 38-Java/src/Caravan.java
public class Caravan extends Veicolo {
    private int potenza;
    
    public Caravan(int codice, int postiLetto, String tipo, String marca, float lunghezza, float larghezza,
            float costoGiornaliero, int potenza) {
        super(codice, postiLetto, tipo, marca, lunghezza, larghezza, costoGiornaliero);
        this.potenza = potenza;
    }

    public String toString(){
        return super.toString() + "\t-\t-\t" + potenza + "\t" + costoGiornaliero;
    }
}
